### PR TITLE
Add CA.F sample showing an all-custom type and its callback.

### LINKED ISSUE #1102: How do we create ErrorHandler in retrofit 2.0?
Hi
It seems ErrorHandler is removed in 2.0 and we need to use CallAdapter instead. Is there any examples to look at?

Cheers,

### PR DESCRIPTION
**Pull request by JakeWharton**
This also remedies some code and configuration problems which prevented the previous one for ListenableFuture from working.

For lack of response from the original poster, closes #1102.

### CODE REVIEW

File: samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import retrofit.Retrofit;
+import retrofit.http.GET;
+
+/**
+ * A sample showing a custom {@link CallAdapter} which adapts the built-in {@link Call} to a custom
+ * version whose callback has more granular methods.
+ */
+public final class ErrorHandlingCallAdapter {
+  /** A callback which offers granular callbacks for various conditions. */
+  interface MyCallback<T> {
```
Comment (JakeWharton) [inline samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java]: Fun fact: this was the built-in Callback API when I first started working on the project in 2012: https://github.com/square/retrofit/blob/d8fab22ab289f24caf73e2af9d486b92afdad490/core/src/main/java/retrofit/core/Callback.java

**Comment by JakeWharton** [comment]
This is based on #1276 which has an API change. Will rebase once that's merged as I'm too lazy to deal with the conflict churn.

File: samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        }
+
+        @Override public <R> MyCall<R> adapt(Call<R> call) {
+          return new MyCallWrapper<>(call);
+        }
+      };
+    }
+  }
+
+  static class MyCallWrapper<T> implements MyCall<T> {
```
Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java]: Wrapper –> Adapter?

File: samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    /** Called for 401 responses. */
+    void unauthenticated(Response<?> response);
+    /** Called for [400, 500) responses, except 401. */
+    void clientError(Response<?> response);
+    /** Called for [500, 600) response. */
+    void serverError(Response<?> response);
+    /** Called for network errors while making the call. */
+    void networkError(IOException e);
+    /** Called for unexpected errors while making the call. */
+    void unexpectedError(Throwable t);
```
Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/ErrorHandlingCallAdapter.java]: Bravo on this interface, BTW

**Comment by swankjesse** [comment]
LGTM

**Pull request merged**